components/brave_rewards/resources/extension/brave_rewards/background/publisherVisit.ts

```diff
@@ -0,0 +1,227 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// Util
+import rewardsPanelActions from './actions/rewardsPanelActions'
+
+interface GreaselionErrorResponse {
+  errorMessage: string
+}
+
+interface SavePublisherVisitResponse {
+  url: string
+  publisherKey: string
+  publisherName: string
+  mediaKey?: string
+  favIconUrl?: string
+}
+
+interface GreaselionResponse {
+  type: 'GreaselionError' | 'SavePublisherVisit'
+  mediaType: string
+  data: GreaselionErrorResponse | SavePublisherVisitResponse | null
+}
+
+interface MediaDurationHandlerRegistrationRequest {
+  urlRegex: string
+}
+
+interface MediaDurationMetadataResponse {
+  mediaKey: string
+  duration: number
+}
+
+// Media duration handler registration sentinel
+interface MediaDurationHandlerRegistration {
+  registered: boolean
+}
+
+// Maps a tabId to a registration sentinel
+const mediaDurationHandlerRegistrations = new Map<number, MediaDurationHandlerRegistration>()
+
+// Maps a tabId to a specific Greaselion port
+const greaselionPorts = new Map<number, chrome.runtime.Port>()
+
+// Maps a media key to a specific publisher
+const publisherKeys = new Map<string, string>()
+
+const handleGreaselionErrorResponse = (tabId: number, mediaType: string, data: GreaselionErrorResponse) => {
+  console.error(`Greaselion error: ${data.errorMessage}`)
+}
+
+const handleMediaDurationMetadataResponse = (tabId: number, mediaType: string, data: MediaDurationMetadataResponse) => {
+  const publisherKey = publisherKeys[data.mediaKey]
+  if (!publisherKey) {
+    console.error(`Failed to handle media duration metadata response: missing publisher key for media key ${data.mediaKey}`)
```

Comment:
can you confirm these errors aren't logged to disk by default? IIRC they aren't.

Reply:
@NejcZdovc Do you know how this one works? I do see these entries in my debug log.

Reply:
this will not be logged on disk in rewards logs